### PR TITLE
code refactoring and fixes

### PR DESCRIPTION
**Pull request by aitezazimtiaz**

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npm run lint
```
Comment (umar0932) [inline .husky/pre-commit:4]: add end of line

File: src/components/AutoComplete/AutoCompleteAcc.tsx [hunk 2]
```diff
@@ -1,47 +1,44 @@
-import React from "react";
-import CurrentLocation from "../../../public/assets/current_location.svg";
-import BrowserLocations from "../../../public/assets/browse_online.svg";
-import StaticImage from "next/image";
-import Link from "next/link";
+import React from 'react'
+import CurrentLocation from '../../../public/assets/current_location.svg'
+import BrowserLocations from '../../../public/assets/browse_online.svg'
+import StaticImage from 'next/image'
+import Link from 'next/link'
```
Comment (umar0932) [inline src/components/AutoComplete/AutoCompleteAcc.tsx:6]: Sort above imports in lexical order

File: src/components/Cards/FollowCard.tsx [hunk 3]
```diff
@@ -1,30 +1,26 @@
...
+import StaticImage from 'next/image'
 
 interface FollowCardProps {
-  title: string;
-  SubTitle: string;
-  imageSrc: string;
+  title: string
+  SubTitle: string
+  imageSrc: string
 }
```
Comment (umar0932) [inline src/components/Cards/FollowCard.tsx:7]: move all the interfaces in a separate file

File: src/components/Cards/OrderCard.tsx [hunk 4]
```diff
@@ -1,31 +1,24 @@
-import Image from "next/image";
-import OrderImage from "../../../public/assets/order_image.png";
+import Image from 'next/image'
+import OrderImage from '../../../public/assets/order_image.png'
 export default function OrderCard() {
```
Comment (umar0932) [inline src/components/Cards/OrderCard.tsx:3]: make it arrow function and give space above

File: src/components/Checkout/Checkout.tsx [hunk 5]
```diff
@@ -1,81 +1,81 @@
...
-      Ref.current = timerId;
+      updateTimer()
+      const timerId = setInterval(updateTimer, 1000)
+      Ref.current = timerId
 
       return () => {
         // Clean up the timer when the component unmounts or isOpen becomes false
         if (Ref.current !== null) {
-          clearInterval(Ref.current as number); // Cast to number
+          clearInterval(Ref.current as number) // Cast to number
```
Comment (umar0932) [inline src/components/Checkout/Checkout.tsx:49]: remove brackets and make it a single line

File: src/components/Forms/AccountSettingsForm/ChangeEmailForm.tsx [hunk 6]
```diff
@@ -1,96 +1,88 @@
...
+      confirm_email: ''
     },
-    onSubmit: (values) => {
-      JSON.stringify(values, null, 2);
-    },
-  });
+    onSubmit: values => {
+      JSON.stringify(values, null, 2)
+    }
+  })
```
Comment (umar0932) [inline src/components/Forms/AccountSettingsForm/ChangeEmailForm.tsx:15]: remove brackets

File: src/components/Forms/AccountSettingsForm/ChangePasswordForm.tsx [hunk 7]
```diff
@@ -1,64 +1,62 @@
...
+      new_password: '',
+      confirm_password: ''
     },
-    onSubmit: (values) => {
-      JSON.stringify(values, null, 2);
-    },
-  });
+    onSubmit: values => {
+      JSON.stringify(values, null, 2)
+    }
```
Comment (umar0932) [inline src/components/Forms/AccountSettingsForm/ChangePasswordForm.tsx:11]: remove brackets

File: src/components/Forms/AccountSettingsForm/EmailPreferanceForm.tsx [hunk 8]
```diff
@@ -1,119 +1,111 @@
-import { Button } from "@/core/ui/button";
-import { Checkbox } from "@/core/ui/checkbox";
+import { Button } from '@/core/ui/button'
+import { Checkbox } from '@/core/ui/checkbox'
 export default function EmailPreferanceForm() {
```
Comment (umar0932) [inline src/components/Forms/AccountSettingsForm/EmailPreferanceForm.tsx:3]: make an arrow function every where

File: src/components/Forms/BasicInfoForm.tsx [hunk 9]
```diff
@@ -1,51 +1,51 @@
...
+import { useState } from 'react'
 
 export default function BasicInfoForm() {
-  const [value, setValue] = useState<any[]>([]);
-  const [charCount, setCharCount] = useState(0);
+  const [value, setValue] = useState<any[]>([])
+  const [charCount, setCharCount] = useState(0)
   // const [tagCharCount, setTagCharCount] = useState(0);
 
   // const [TagCharCount, setTagCharCount] = useState(0);
```
Comment (umar0932) [inline src/components/Forms/BasicInfoForm.tsx:37]: remove comments

File: src/components/Forms/BasicInfoForm.tsx [hunk 10]
```diff
@@ -1,51 +1,51 @@
...
-      setValue(newValue);
-      setTagCount(newValue.length);
+      setValue(newValue)
+      setTagCount(newValue.length)
     }
-  };
+  }
 
   // const maxTagLimit = 25;
   // const [tags, setTags] = useState([]);
```
Comment (umar0932) [inline src/components/Forms/BasicInfoForm.tsx:51]: remove comment

File: src/components/Forms/BasicInfoForm.tsx [hunk 11]
```diff
@@ -68,12 +68,12 @@ export default function BasicInfoForm() {
...
-      formik.handleChange(e); // Handle formik changes
-      setCharCount(inputValue.length); // Update charCount
+      formik.handleChange(e) // Handle formik changes
+      setCharCount(inputValue.length) // Update charCount
     }
-  };
+  }
 
   // const handleTagChange = (e: any) => {
   //   const tagValue = e.target.value;
```
Comment (umar0932) [inline src/components/Forms/BasicInfoForm.tsx:79]: remove comment

File: src/components/Forms/BasicInfoForm.tsx [hunk 12]
```diff
@@ -85,113 +85,107 @@ export default function BasicInfoForm() {
...
-                  placeholder="Tell Attendees who is organizing this event"
+                  className='h-[25px] bg-transparent outline-none placeholder:text-sm'
+                  id='organizer_name'
+                  name='organizer_name'
+                  type='text'
+                  placeholder='Tell Attendees who is organizing this event'
                   onChange={formik.handleChange}
                   value={formik.values.organizer_name}
                 />
               </div>
```
Comment (umar0932) [inline src/components/Forms/BasicInfoForm.tsx:148]: make a separate component for input and import it from there

File: src/components/Forms/BasicInfoForm.tsx [hunk 13]
```diff
@@ -85,113 +85,107 @@ export default function BasicInfoForm() {
...
+                <span className='text-xl font-bold text-primary'>Tags</span>
+                <span className='text-md text-[#757575] max-sm:text-sm'>
+                  Improve discoverability of your event by adding tags relevant to the subject
+                  matter.
                 </span>
-                <div className=" flex  flex-col flex-wrap rounded-lg bg-[#EFF8FF] p-3 ">
+                <div className=' flex  flex-col flex-wrap rounded-lg bg-[#EFF8FF] p-3 '>
                   <span>Add Tags</span>
                   {/* <ReactSelect
                     options={options}
```
Comment (umar0932) [inline src/components/Forms/BasicInfoForm.tsx:191]: remove comment

File: src/components/Forms/BasicInfoForm.tsx [hunk 14]
```diff
@@ -213,13 +207,13 @@ export default function BasicInfoForm() {
                     isMulti
                     isOptionDisabled={() => value.length >= 10}
                     options={options}
-                    placeholder="Add search key"
+                    placeholder='Add search key'
                     styles={{
-                      control: (baseStyles, state) => ({
+                      control: (baseStyles) => ({
```
Comment (umar0932) [inline src/components/Forms/BasicInfoForm.tsx:212]: remove brackets

File: src/components/Forms/CheckoutForm.tsx [hunk 15]
```diff
@@ -1,71 +1,66 @@
...
+      last_name: '',
+      email: ''
     },
-    onSubmit: (values) => {
-      JSON.stringify(values, null, 2);
-    },
-  });
+    onSubmit: values => {
+      JSON.stringify(values, null, 2)
+    }
```
Comment (umar0932) [inline src/components/Forms/CheckoutForm.tsx:19]: remove brackets

File: src/components/Forms/CheckoutForm.tsx [hunk 16]
```diff
@@ -74,100 +69,93 @@ export default function CheckoutForm() {
...
                 />
               </div>
-              <div className=" flex items-center justify-between gap-5 border-t border-t-[#9CB2DF] bg-white p-5">
+              <div className=' flex items-center justify-between gap-5 border-t border-t-[#9CB2DF] bg-white p-5'>
                 <Label
-                  htmlFor="option-two"
-                  className=" text-md flex items-center gap-2 text-[#313131] max-sm:text-sm"
+                  htmlFor='option-two'
+                  className=' text-md flex items-center gap-2 text-[#313131] max-sm:text-sm'
                 >
```
Comment (umar0932) [inline src/components/Forms/CheckoutForm.tsx:101]: make a separate component for label

File: src/components/Forms/CheckoutForm.tsx [hunk 17]
```diff
@@ -74,100 +69,93 @@ export default function CheckoutForm() {
...
-                  <Image
-                    src={PayPalIcon}
-                    alt="pay-pal-icon"
-                    className="h-[25px] w-[25px]"
-                  />
```
Comment (umar0932) [inline src/components/Forms/CheckoutForm.tsx:113]: make a separate component for image

File: src/components/Forms/CheckoutForm.tsx [hunk 18]
```diff
@@ -74,100 +69,93 @@ export default function CheckoutForm() {
...
+                  <Image src={PayPalIcon} alt='pay-pal-icon' className='h-[25px] w-[25px]' />
                   PayPal
                 </Label>
                 <RadioGroupItem
-                  value="option-two"
-                  id="option-two"
-                  className="h-[20px] w-[20px] border-btnprimary bg-white text-btnprimary"
+                  value='option-two'
+                  id='option-two'
+                  className='h-[20px] w-[20px] border-btnprimary bg-white text-btnprimary'
```
Comment (umar0932) [inline src/components/Forms/CheckoutForm.tsx:108]: make a separate component for RadioGroupItem

File: src/components/Forms/CheckoutForm.tsx [hunk 19]
```diff
@@ -74,100 +69,93 @@ export default function CheckoutForm() {
...
-              <span className="text-[#757575]">
-                Keep me updated on more events and news from this event
-                organizer
+              <span className='text-[#757575]'>
+                Keep me updated on more events and news from this event organizer
               </span>
             </div>
-            <div className="flex items-center space-x-2">
+            <div className='flex items-center space-x-2'>
               <Checkbox defaultChecked={true} />
```
Comment (umar0932) [inline src/components/Forms/CheckoutForm.tsx:124]: make separate component of checkbox

File: src/components/Forms/CheckoutForm.tsx [hunk 20]
```diff
@@ -74,100 +69,93 @@ export default function CheckoutForm() {
...
-              <Button className="text-md mt-3 h-[45px] w-[150px] bg-btnprimary font-bold text-white hover:bg-btnsecondary">
+            <Link href='/'>
+              <Button className='text-md mt-3 h-[45px] w-[150px] bg-btnprimary font-bold text-white hover:bg-btnsecondary'>
                 Register
               </Button>
             </Link>
           </div>
         </div>
 
         {/* Right Side */}
```
Comment (umar0932) [inline src/components/Forms/CheckoutForm.tsx:140]: remove comment

File: src/components/Forms/ContactSalesForm.tsx [hunk 21]
```diff
@@ -1,220 +1,206 @@
...
+              id='company_name'
+              name='company_name'
+              type='text'
               onChange={formik.handleChange}
               value={formik.values.company_name}
-              style={{ boxShadow: "0px 0px 4px 0px rgba(21, 86, 173, 0.55)" }}
+              style={{ boxShadow: '0px 0px 4px 0px rgba(21, 86, 173, 0.55)' }}
             />
           </div>
           {/* First and Last Name */}
```
Comment (umar0932) [inline src/components/Forms/ContactSalesForm.tsx:57]: remove comment

File: src/components/Forms/ContactSalesForm.tsx [hunk 22]
```diff
@@ -1,220 +1,206 @@
...
           </div>
           {/* Company Name */}
-          <div className="flex flex-col gap-3">
-            <span className="text-base text-[#757575]">
-              Company name<span className="text-[#F00]">*</span>
+          <div className='flex flex-col gap-3'>
+            <span className='text-base text-[#757575]'>
+              Company name<span className='text-[#F00]'>*</span>
             </span>
             <input
```
Comment (umar0932) [inline src/components/Forms/ContactSalesForm.tsx:47]: use custom Input component everywhere

File: src/components/Forms/ContactSalesForm.tsx [hunk 23]
```diff
@@ -1,220 +1,206 @@
...
               </SelectTrigger>
               <SelectContent>
                 <SelectGroup>
-                  <SelectLabel className=" font-normal">
-                    United State
-                  </SelectLabel>
+                  <SelectLabel className=' font-normal'>United State</SelectLabel>
                 </SelectGroup>
               </SelectContent>
             </Select>
```
Comment (umar0932) [inline src/components/Forms/ContactSalesForm.tsx:160]: make a separate component for select

File: src/components/Forms/DateTimeForm.tsx [hunk 24]
```diff
@@ -1,141 +1,134 @@
...
-  const [activeButton, setActiveButton] = useState("SingleEvent");
+  const [activeButton, setActiveButton] = useState('SingleEvent')
 
   const handleButtonClick = (buttonName: any) => {
-    setActiveButton(buttonName === activeButton ? null : buttonName);
-  };
+    setActiveButton(buttonName === activeButton ? null : buttonName)
+  }
 
   const buttonStyles = (buttonName: any) => {
```
Comment (umar0932) [inline src/components/Forms/DateTimeForm.tsx:20]: remove brackets

File: src/components/Forms/DateTimeForm.tsx [hunk 25]
```diff
@@ -1,141 +1,134 @@
...
               <button
-                className="h-[55px] w-[180px] rounded-lg bg-[#EFF8FF]"
-                style={buttonStyles("SingleEvent")}
-                onClick={() => handleButtonClick("SingleEvent")}
+                className='h-[55px] w-[180px] rounded-lg bg-[#EFF8FF]'
+                style={buttonStyles('SingleEvent')}
+                onClick={() => handleButtonClick('SingleEvent')}
               >
                 Single Event
               </button>
```
Comment (umar0932) [inline src/components/Forms/DateTimeForm.tsx:51]: make separate button component everywhere

File: src/components/Forms/GeneralInfoForm.tsx [hunk 26]
```diff
@@ -1,100 +1,94 @@
...
+      JSON.stringify(values, null, 2)
+    }
+  })
   return (
     <>
-      <div className="w-full items-center justify-center ">
-        <span className="flex justify-center text-3xl font-bold text-primary">
-          General Info
-        </span>
-        <form
```
Comment (umar0932) [inline src/components/Forms/GeneralInfoForm.tsx:23]: make separate form component

File: src/components/Forms/OrganizerProfileForm.tsx [hunk 27]
```diff
@@ -1,143 +1,130 @@
...
+          <Button className='flex h-[43px] w-[120px] items-center justify-center rounded-sm bg-[#DEF0FF] p-3 font-bold text-primary'>
             Cancel
           </Button>
-          <Link href="/organizer">
-            <Button className="flex h-[43px] w-[120px] items-center justify-center rounded-sm bg-btnprimary p-3 font-bold text-white hover:bg-btnsecondary">
+          <Link href='/organizer'>
+            <Button className='flex h-[43px] w-[120px] items-center justify-center rounded-sm bg-btnprimary p-3 font-bold text-white hover:bg-btnsecondary'>
               Save
             </Button>
           </Link>
```
Comment (umar0932) [inline src/components/Forms/OrganizerProfileForm.tsx:125]: make Link component separate

File: src/components/HomePage/InterestsDialog.tsx [hunk 28]
```diff
@@ -1,65 +1,60 @@
...
-      setSelected(selected.filter((item) => item !== interest));
+      setSelected(selected.filter(item => item !== interest))
     } else {
-      setSelected([...selected, interest]);
+      setSelected([...selected, interest])
     }
-  };
+  }
 
   const handleClose = () => {
```
Comment (umar0932) [inline src/components/HomePage/InterestsDialog.tsx:33]: remove brackets

File: src/components/Cards/EventCard.tsx [hunk 29]
```diff
@@ -1,74 +1,63 @@
...
+  event: Event // Use the Event interface for the event prop
 }
 
 export default function EventCard({ event }: EventCardProps) {
-  const [isLiked, setIsLiked] = useState(false);
+  const [isLiked, setIsLiked] = useState(false)
 
   const toggleLike = () => {
-    setIsLiked(!isLiked);
-  };
```
Comment (umar0932) [inline src/components/Cards/EventCard.tsx:26]: remove brackets

File: src/components/Forms/AccountSettingsForm/PaymentCardForm.tsx [hunk 30]
```diff
@@ -1,124 +1,122 @@
...
+import { useFormik } from 'formik'
+import { Button } from '@/core/ui/button'
 
 export default function PaymentCardForm() {
-  const maxCardLength = 12;
-  const maxCVVLength = 3;
+  const maxCardLength = 12
+  const maxCVVLength = 3
 
   const formatCardNumber = (input: any) => {
```
Comment (umar0932) [inline src/components/Forms/AccountSettingsForm/PaymentCardForm.tsx:9]: remove bracket

**Review by umar0932 — CHANGES_REQUESTED** [review]

File: src/components/Forms/AccountSettingsForm/ChangeEmailForm.tsx (hunk 6 above)
Comment (aitezazimtiaz) [reply src/components/Forms/AccountSettingsForm/ChangeEmailForm.tsx:16]: error on removing brackets

**Review by umar0932 — APPROVED** [review]

**Pull request merged**